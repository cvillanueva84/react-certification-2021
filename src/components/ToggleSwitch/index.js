export { default } from './TogleSwitch.component';
