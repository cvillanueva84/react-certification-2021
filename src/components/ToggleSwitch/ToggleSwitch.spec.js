import ToggleSwitch from './TogleSwitch.component';

const { render, screen } = require('@testing-library/react');

describe('ToggleSwitch', () => {
  let container;

  beforeEach(() => {
    container = render(<ToggleSwitch />).container;
  });
  test('Should renders a ToggleStwitch component', () => {
    expect(container).toMatchSnapshot();
  });

  test('Should contains a label with a switch role', () => {
    const labelElement = screen.getByRole('switch');
    expect(labelElement).toBeInTheDocument();
  });
  test('Should contains an input type checkbox', () => {
    const inputElement = screen.getByRole('button');
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('type', 'checkbox');
  });
  test('Should contains a span tag', () => {
    const spanElement = container.querySelector('span');
    expect(spanElement).toBeInTheDocument();
  });
});
