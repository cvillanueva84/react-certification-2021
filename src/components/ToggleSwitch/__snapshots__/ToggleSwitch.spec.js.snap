// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch Should renders a ToggleStwitch component 1`] = `
<div>
  <label
    class="sc-bdnxRM fzaqfv switch"
    for="switch-dark-mode"
    role="switch"
  >
    <input
      class="sc-gtsrHT flAbm"
      id="switch-dark-mode"
      role="button"
      type="checkbox"
    />
    <span
      class="sc-dlnjwi dSfqLy"
    />
  </label>
</div>
`;
